File: src/main/java/no/fint/provider/personnel/behaviour/Behaviour.java
package no.fint.provider.personnel.behaviour;

import no.fint.event.model.Event;
import no.fint.event.model.Problem;
import org.jooq.lambda.function.Consumer2;

import java.util.List;
import java.util.Objects;

public interface Behaviour<T> extends Consumer2<Event,T> {
    default Problem newProblem(String field, String message) {
        Problem problem = new Problem();
        problem.setField(field);
        problem.setMessage(message);
        return problem;
    }

    default boolean empty(List<?> l) {
        return Objects.isNull(l) || l.isEmpty();
    }
}

File: src/main/kotlin/no/fint/provider/springer/behaviour/Behaviour.kt
package no.fint.provider.springer.behaviour

import no.fint.event.model.Event
import no.fint.event.model.Problem
import java.util.*

fun interface Behaviour<T> {
    fun accept(event: Event<*>, payload: T)

    fun addProblem(event: Event<*>, field: String, message: String) {
        if (event.problems == null) {
            event.problems = ArrayList()
        }
        val problem = Problem().apply {
            this.field = field
            this.message = message
        }
        event.problems.add(problem)
    }

    fun empty(list: List<*>?): Boolean = list.isNullOrEmpty()
}
